src/lib/components/Avatar/Avatar.tsx

```diff
@@ -72,19 +63,38 @@ const AvatarImage = styled.img`
 `;
 
 export function Avatar({
-  name,
-  size = "extra large",
+  initials,
+  size = "extraLarge",
   image,
   status,
 }: AvatarProps) {
-  const [firstName, surname] = name.split(" ");
-  const initials = (firstName[0] + (surname ? surname[0] : "")).toUpperCase();
+  const [isImageValid, setIsImageValid] = useState(false);
+
+  useEffect(() => {
+    if (image) {
+      new Promise<boolean>((resolve, reject) => {
+        const loadImg = new Image();
+        loadImg.src = image;
+        loadImg.onload = () => {
+          return resolve(loadImg.complete && loadImg.naturalWidth > 1);
+        };
+        loadImg.onerror = () => {
+          return reject(false);
+        };
+      })
+        .then(isValid => {
+          setIsImageValid(isValid);
+        })
+        .catch(() => {
+          setIsImageValid(false);
+        });
+    }
+  }, [image]);
+
```

Comment:
We can use built in the `img` tag callbacks `onLoad` & `onError`.
This component could look like: 
```TypeScript
export function Avatar({
  initials,
  size = "extraLarge",
  image,
  status,
  alt = "Avatar",
}: AvatarProps) {
  const [isImageLoaded, setImageLoaded] = useState(false);

  const handleImageLoad = () => setImageLoaded(true);
  const handleImageError = () => setImageLoaded(false);

  return (
    <StyledAvatar size={size}>
      <AvatarImage
        src={image}
        onLoad={handleImageLoad}
        onError={handleImageError}
        style={{ display: isImageLoaded ? "block" : "none" }}
        alt={alt}
      />
      {!isImageLoaded && initials}
      {status && <StatusIndicatior status={status} role="status" />}
    </StyledAvatar>
  );
}
```